src/modules/jcms_article/src/Hooks/NodePresave.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Drupal\jcms_article\Hooks;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\jcms_article\FetchArticleVersions;
+use Drupal\paragraphs\Entity\Paragraph;
+
+/**
+ * Class NodePresave.
+ *
+ * @package Drupal\jcms_article
+ * @todo Look to share more code with \Drupal\jcms_article\ArticleCrud.
+ */
+final class NodePresave {
+
+  /**
+   * @var \Drupal\jcms_article\FetchArticleVersions
+   */
+  private $fetchArticleVersions;
+
+  /**
+   * @var string
+   */
+  private static $articleData;
+
+  /**
+   * NodePresave constructor.
+   *
+   * @param \Drupal\jcms_article\FetchArticleVersions $fetch_article_versions
+   */
+  public function __construct(FetchArticleVersions $fetch_article_versions) {
+    $this->fetchArticleVersions = $fetch_article_versions;
+  }
+
+  /**
+   * Gets the article data.
+   *
+   * @param $id
+   *
+   * @return \Drupal\jcms_article\Entity\ArticleVersions|null
+   */
+  public function getArticleById($id) {
+    if (!self::$articleData) {
+      self::$articleData = $this->fetchArticleVersions->getArticleVersions($id);
+    }
+    return self::$articleData;
+  }
+
+  /**
+   * Adds the JSON fields to the node.
+   */
+  public function addJsonFields(EntityInterface $entity) {
+    if ($entity->get('field_article_json')->getValue()) {
+      $this->updateJsonParagraph($entity);
+    }
+    else {
+      $this->createJsonParagraph($entity);
+    }
+  }
+
+  /**
+   * Sets the published date on the node.
+   */
+  public function setPublishedDate(EntityInterface $entity) {
+    $id = $entity->label();
+    $article = $this->getArticleById($id);
+    // Use the published JSON if no unpublished exists.
+    $version = $article->getLatestUnpublishedVersionJson() ?: $article->getLatestPublishedVersionJson();
+    $json = json_decode($version);
+    if (property_exists($json, 'versionDate')) {
+      $date = strtotime($json->versionDate);
+      $entity->set('created', $date);
```

Comment:
The created date should be set to the published date of an article which is the same for all versions. This will meet the requirement in ELPP-1839.

There is an additional requirement that can be handled in a separate jira ticket to store the statusDate of the latest published version of an article which can be used for sorting appropriately.